src/shortestpaths/bellman-ford.jl

```diff
@@ -115,7 +115,7 @@ function enumerate_paths(state::AbstractPathState, vs::Vector{T}) where T<:Integ
     all_paths = Vector{Vector{T}}(undef, num_vs)
     for i = 1:num_vs
         all_paths[i] = Vector{T}()
-        index = vs[i]
+        index::T = vs[i]
```

Comment:
Why do we need this type assertion?

Reply:
You are right, it is better to write `currepathindex = U(i)` here

Reply:
@simonschoelly Could you make these last few changes? I'm ready for final review and merge so we don't get backed up too much. Thanks.

Reply:
I must have forgotten this one, should be fixed now.